handlers/workloads.go

```diff
@@ -93,6 +96,11 @@ func WorkloadDetails(w http.ResponseWriter, r *http.Request) {
 
 	criteria := business.WorkloadCriteria{Namespace: p.Namespace, WorkloadName: p.WorkloadName, WorkloadType: p.WorkloadType, IncludeIstioResources: true, IncludeServices: true, IncludeHealth: p.IncludeHealth, RateInterval: p.RateInterval, QueryTime: p.QueryTime, Cluster: p.Cluster}
 
+	cluster := criteria.Cluster
+	if cluster == "null" || cluster == "" {
```

Comment:
Maybe we could create a generic funcion, like isValidCluster, so it can be used by other handlers and also validates that the cluster exist and avoid to propagate the error to other layers. But this also be addressed later

Reply:
Agree, but let's have a separate PR to cleanup this kind of things.